app/com/lynxanalytics/biggraph/frontend_operations/Operations.scala

```diff
@@ -178,55 +178,63 @@ abstract class ProjectOperations(env: SparkFreeEnvironment) extends OperationReg
       throw new AssertionError(s"Unexpected type (${attr.typeTag}) on $attr")
   }
 
-  def parseAggregateParams(params: ParameterHolder) = {
+  def parseAggregateParams(params: ParameterHolder, prefix: String = "", weight: String = null) = {
+    val extraSuffix = if (weight == null) "" else s"_by_$weight"
+    val addSuffix = params("add_suffix") == "yes"
     val aggregate = "aggregate_(.*)".r
     params.toMap.toSeq.collect {
       case (aggregate(attr), choices) if choices.nonEmpty => attr -> choices
     }.flatMap {
-      case (attr, choices) => choices.split(",", -1).map(attr -> _)
+      case (attr, choices) =>
+        assert(addSuffix || !choices.contains(","),
+          s"Suffixes are necessary when multiple aggregations are configured for $attr.")
+        choices.split(",", -1).map {
+          c => (attr, c, if (addSuffix) s"$prefix${attr}_$c$extraSuffix" else s"$prefix$attr")
+        }
     }
   }
   def aggregateParams(
     attrs: Iterable[(String, Attribute[_])],
     needsGlobal: Boolean = false,
     weighted: Boolean = false): List[OperationParameterMeta] = {
     val sortedAttrs = attrs.toList.sortBy(_._1)
-    sortedAttrs.toList.map {
-      case (name, attr) =>
-        val options = if (attr.is[Double]) {
-          if (weighted) { // At the moment all weighted aggregators are global.
-            FEOption.list("weighted_average", "by_max_weight", "by_min_weight", "weighted_sum")
-          } else if (needsGlobal) {
-            FEOption.list(
-              "average", "count", "count_distinct", "count_most_common", "first", "max", "min", "most_common",
-              "std_deviation", "sum")
-
+    Choice("add_suffix", "Add suffixes to attribute names", options = FEOption.yesno) +:
+      sortedAttrs.toList.map {
```

Comment:
I didn't change this, it just gets indented by Scalariform. I've only added the line above.